src/modules/role/role.service.ts

```diff
@@ -205,39 +205,50 @@ export class RoleService {
     return verifiedRoles.filter(Boolean);
   }
 
-  public async verifyEnrolmentPrecondition({
-    userDID,
+  /**
+   * Fetches enrolment preconditions for a given role (claim type)
+   * @param claimType the role to fetch enrolment preconditions for
+   * @return enrolment preconditions for a given role (claim type)
+   */
+  public async fetchEnrolmentPreconditions({
     claimType,
   }: {
-    userDID: string;
     claimType: string;
   }) {
-    const [didDocument, role] = await Promise.all([
-      this.didService.getById(userDID),
-      this.getByNamespace(claimType),
-    ]);
-
+    const role = await this.getByNamespace(claimType);
     if (!role) {
       throw new Error(`There is no created role for ${claimType} namespace`);
     }
 
     const {
       definition: { enrolmentPreconditions },
     } = role;
+    return enrolmentPreconditions;
+  }
 
-    if (!enrolmentPreconditions || enrolmentPreconditions.length < 1) return;
-    for (const { type, conditions } of enrolmentPreconditions) {
-      if (type === 'role' && conditions && conditions?.length > 0) {
-        const conditionMet = didDocument.service.some(
-          ({ claimType }) =>
-            claimType && conditions.includes(claimType as string)
-        );
-        if (!conditionMet) {
-          throw new Error(
-            `Role enrolment precondition not met for user: ${userDID} and role: ${claimType}`
-          );
-        }
-      }
+  /**
+   * Verifies that a user's Did Document contains all roles required for enrolment (enrolment preconditions)
+   * @param claimType the role to verify enrollment preconditions for
+   * @param userDID the Did of the user seeking to obtain the role. Must posses enrolment preconditions
+   * @param conditions enrolment preconditions needed to obtain the role (claimType)
+   */
+  public async verifyDidDocumentContainsEnrolmentPreconditions({
```

Comment:
From my perspective verification belongs to `claim` module

Reply:
got it. I moved this to the claim-verification service. Lmk if you think it is not good there.